cmd/check-spelling/kata-spell-check.sh

```diff
@@ -0,0 +1,347 @@
+		sed 's/\[[^]]*\]([^\)]*)//g' || true)
+
+	# Convert the file into "pure ASCII" by removing all awkward
+	# Unicode characters that won't spell check.
+	#
+	# Necessary since pandoc is "clever" and will convert things like
+	# GitHub's colon emojis (such as ":smile:") into the actual utf8
+	# character where possible.
+	utf8_free_doc=$(echo "$pandoc_doc" | iconv -c -f utf-8 -t ascii)
+
+	# Next, perform the following simplifications:
+	#
+	# - Remove URLs.
+	# - Remove email addresses.
+	# - Replace most punctuation symbols with a space
+	#   (excluding a dash (aka hyphen!)
+	# - Carefully remove non-hyphen dashes.
+	# - Remove GitHub @userids.
+	pre_hunspell_doc=$(echo "$utf8_free_doc" |\
+		sed 's,https*://[^[:space:]()][^[:space:]()]*,,g' |\
+		sed -r 's/[a-zA-Z0-9.-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9.-]+//g' |\
+		tr '[,\[\]()\*\\/\|=]' ' ' |\
+		sed -e 's/^ *-//g' -e 's/- $//g' -e 's/ -//g' |\
+		sed 's/@[a-zA-Z0-9][a-zA-Z0-9]*\b//g')
+
+	# Call the spell checker
+	hunspell_results=$(echo "$pre_hunspell_doc" | $spell_check_cmd -d "${dict_languages}")
+
+	# Finally, post-process the hunspell output:
+	#
+	# - Parse the output to ignore:
+	#   - Hunspell banner.
+	#   - Correctly spelt words (lines starting with '*', '+' or '-').
+	#   - All words containing numbers (like "100MB").
+	#   - All words that appear to be acronymns / Abbreviations
+	#     (atleast two upper-case letters and which may be plural or
+	#     possessive).
+	#   - All words that appear to be numbers.
+	#   - All possessives and the dreaded isolated "'s" which occurs
+	#     for input like this:
+	#
+	#         `kata-shim`'s
+	#
+	#     which gets converted by $strip_cmd into simply:
+	#
+	#         's
+	#
+	# - Sort output.
+
+	final_results=$(echo "$hunspell_results" |\
+		grep -Evi "(ispell|hunspell)" |\
+		grep -Ev '^(\*|\+|-)' |\
+		grep -Evi "^(&|#) [^ ]*[0-9][^ ]*" |\
+		grep -Ev "^. [A-Z][A-Z][A-Z]*(s|'s)*" |\
+		grep -Ev "^. 's" |\
+		sort -u || true)
+
+	local line
+	local incorrects
+	local near_misses
+
+	near_misses=$(echo "$final_results" | grep '^&' || true)
+	incorrects=$(echo "$final_results" | grep '^\#' | awk '{print $2}' || true)
+
+	local -i failed=0
+
+	[ -n "$near_misses" ] && failed+=1
+	[ -n "$incorrects" ] && failed+=1
+
+	echo "$near_misses" | while read -r line
+	do
+		[ "$line" = "" ] && continue
+
+		local word
+		local possibles
+
+		word=$(echo "$line" | awk '{print $2}')
+		possibles=$(echo "$line" | cut -d: -f2- | sed 's/^ *//g')
+
+		warn "Word '${word}': did you mean one of the following?: ${possibles}"
+	done
+
+	local incorrect
+	for incorrect in $incorrects
+	do
+		warn "Incorrect word: '$incorrect'"
+	done
+
+	[ "$failed" -gt 0 ] && die "Spell check failed for file: '$file'"
+
+	info "Spell check successful for file: '$file'"
+}
+
+delete_dictionary()
+{
+	rm -f "${KATA_DICT_FILE}" "${KATA_RULES_FILE}"
+}
+
+setup()
+{
+	source /etc/os-release || source /usr/lib/os-release
+
+	# Installing hunspell on SUSE distros pulls in a lot of large
```

Comment:
btw @marcov - have I understood this correctly? Is there a simple solution to avoid making this a NOP for SUSE distros?

Reply:
Where does that comment come from? 
That's true, at least for Leap 15:
```
$ docker run -it --rm opensuse/leap bash -c "zypper -q ref; zypper in -y hunspell"                                                                                                                      
Loading repository data...
Reading installed packages...
Resolving package dependencies...

The following 2 NEW packages are going to be installed:
  hunspell libhunspell-1_6-0
```

Reply:
Aha - I think I got confused by the dictionary installation part. If you install hunspell on SUSE, it doesn't actually work since there is no dependency on any packages that provide dictionaries :smile:...

```sh
$ rpm -q hunspell
hunspell-1.6.2-lp150.1.5.x86_64
$ echo hello|hunspell -a
Can't open affix or dictionary files for dictionary named "en_US".
$ echo $?
1
```

I thought you needed to install the libreoffice langpack packages to get the dictionaries, but seemingly you don't - you need to install the myspell packages instead.

Sorry for the confusion - branch updated!